src/views/activity/activity.view.tsx

```diff
@@ -48,19 +49,26 @@ const Activity: FC = () => {
   const onDisplayPending = () => setDisplayAll(false);
 
   const onClaim = (bridge: Bridge) => {
-    setDisabledBridges((current) => [...current, bridge.id]);
     if (bridge.status === "on-hold") {
+      setDisabledBridges([...disabledBridges, bridge.id]);
       claim({
         bridge,
       })
         .then(() => {
+          if (
+            bridgeList.status === "successful" ||
+            bridgeList.status === "reloading" ||
+            bridgeList.status === "loading-more-items"
+          ) {
+            processFetchBridgesSuccess(bridgeList.data);
+          }
           openSnackbar({
             type: "success-msg",
-            text: "Transaction successfully submitted.\nThe list will be updated once it is processed.",
+            text: "Transaction successfully submitted.",
           });
         })
         .catch((error) => {
-          setDisabledBridges((current) => current.filter((id) => id !== bridge.id));
+          setDisabledBridges(disabledBridges.filter((id) => id !== bridge.id));
```

Comment:
Why this change? Isn't this introducing a bug when we have already some bridges disabled?

Reply:
Nope, why would the number of bridges disabled affect the behavior of this `setDisabledBridges`?